spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,105 @@
+        :data_type                      => :string,
+        :excluded_by_preprocess_options => false,
+        :include                        => {},
+        :tag                            => true,
+        :virtual_column                 => false,
+        :virtual_reflection             => false
+      )
+    end
+
+    it "return column info for model.association.managed-field" do
+      field = "Vm.host.managed-environment"
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => :string,
+        :excluded_by_preprocess_options => false,
+        :include                        => {},
+        :tag                            => true,
+        :virtual_column                 => false,
+        :virtual_reflection             => false
+      )
+    end
+
+    it "return column info for model-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm-cpu_limit"
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => :integer,
+        :excluded_by_preprocess_options => false,
+        :format_sub_type                => :integer,
+        :include                        => {},
+        :tag                            => false,
+        :virtual_column                 => false,
+        :virtual_reflection             => false
+      )
+      expect(col_info).to eq(described_class.get_col_info("Vm-cpu_limit"))
+    end
+
+    it "return column info for model.association-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm.guest_applications-vendor"
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => :string,
+        :excluded_by_preprocess_options => false,
+        :format_sub_type                => :string,
+        :include                        => {:guest_applications => {}},
+        :tag                            => false,
+        :virtual_column                 => false,
+        :virtual_reflection             => false
+      )
+      expect(col_info).to eq(described_class.get_col_info("Vm.guest_applications-vendor"))
+    end
+
+    it "return column info for model.virtualassociation..virtualassociation-field" do
+      # model.virtualassociation.virtualassociation.
+      field = "ManageIQ::Providers::InfraManager::Vm.service.user.vms-active"
+      col_info = described_class.get_col_info(field)
+      expect(col_info).to match(
+        :data_type                      => :boolean,
+        :excluded_by_preprocess_options => false,
+        :format_sub_type                => :boolean,
+        :include                        => {},
+        :tag                            => false,
+        :virtual_column                 => true,
+        :virtual_reflection             => true,
+      )
+      expect(col_info).to eq(described_class.get_col_info("Vm.service.user.vms-active"))
```

Comment:
It is testing parsing.
I think entry point to this method could be "ManageIQ::Providers::InfraManager::Vm.service.user.vms-active"
OR
"Vm.service.user.vms-active"